[Dexter Sim] iP

dexter-sim opened the pull request:
# Duke

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Duke is a task tracking software that is
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is
1. download it from [here](https://github.com/dexter-sim/ip).
2. double click it.
3. add tasks.
4. let it manage your tasks for you 😄 

And it is **FREE**!

Features
- [x] Manage tasks
- [x] Manage deadlines
- [ ] Reminders (coming soon) 
---
If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

dexter-sim committed:
Implement skeletal version of Duke

dexter-sim committed:
Add the ability to store and display text

dexter-sim committed:
Add the ability to mark tasks as done

dexter-sim committed:
Add support for tracking three types of tasks

dexter-sim committed:
Add support for tracking three types of tasks

dexter-sim committed:
Add automated text ui testing

dexter-sim committed:
Add the ability to handle incorrect user inputs

dexter-sim committed:
Add support for deleting tasks from list

dexter-sim committed:
Add file io

dexter-sim committed:
Add local date

dexter-sim committed:
Merge branch 'branch-level-7'

dexter-sim committed:
Merge branch 'branch-level-8'

dexter-sim committed:
Add more oop

dexter-sim committed:
Clean up files

dexter-sim committed:
Bug fix for file io

dexter-sim committed:
Refactor exceptions

dexter-sim committed:
Handle io exceptions with duke exception

dexter-sim committed:
Handle io exceptions with duke exception

dexter-sim committed:
Bug fix

dexter-sim committed:
Change handling of command exceptions

dexter-sim committed:
Reorganize oop methods

dexter-sim committed:
Update automated testing expected file

dexter-sim committed:
Divide classes into packages

dexter-sim committed:
Fix exception message

dexter-sim committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

dexter-sim committed:
Change main class name for gradle

dexter-sim committed:
Refactor adding tasks to tasklist

dexter-sim committed:
Add JUnit tests

dexter-sim committed:
Add JavaDoc comments to the code

dexter-sim committed:
Tweak the code to comply with a coding standard

dexter-sim committed:
Add the ability to find a task with a keyword

dexter-sim committed:
Add JavaDoc for classes and methods for find task

dexter-sim committed:
Merge branch 'branch-A-JavaDoc'

dexter-sim committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/command/Command.java

dexter-sim committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/task/Task.java

dexter-sim committed:
Setup gradle with javafx

dexter-sim committed:
Add javafx part 1 and 2

dexter-sim committed:
Add javafx part 3

dexter-sim committed:
Add javafx part 4

dexter-sim committed:
Integrate javafx with main duke app

Tan-Jia-Rong commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,75 @@
+     * @return A string corresponding to the task.
+     */
+    public String stringify() {
+        return String.format("%d | %s", isDone ? 1 : 0, description);
```
Can afford to improve code quality  by avoiding complicated expressions
Personally, I feel that it would be better to compute the predicate before putting it into the return statement for clarity
```suggestion
        int doneStatus = isDone ? 1 : 0;
        return String.format("%d | %s", doneStatus, description);
```

Tan-Jia-Rong commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,75 @@
+     * @return A string corresponding to the task.
+     */
+    public String stringify() {
+        return String.format("%d | %s", isDone ? 1 : 0, description);
```
Can afford to improve code quality  by avoiding complicated expressions
Personally, I feel that it would be better to compute the predicate before putting it into the return statement for clarity
```suggestion
        int doneStatus = isDone ? 1 : 0;
        return String.format("%d | %s", doneStatus, description);
```

dexter-sim replied:
Thank you for the suggestion. I like your suggestion and I have committed your suggested change.

Tan-Jia-Rong reviewed (changes requested):
LGTM! :+1:

Just some nits to pick but overall, I found your code to be quite readable and the association between classes are also quite logical
I've only managed to find a minor details that I felt would help to improve the code quality. However aside from that all is well 😄

dexter-sim committed:
Refactor commands for javafx

dexter-sim committed:
Merge branch 'branch-Level-10'

dexter-sim committed:
Add checkStyle

dexter-sim committed:
Update src/main/java/duke/task/Task.java

Co-authored-by: Jia Rong <71122323+Tan-Jia-Rong@users.noreply.github.com>

dexter-sim committed:
Add newline character to error message

dexter-sim committed:
Merge branch 'branch-A-CheckStyle'

njxue commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,107 @@
+     * @throws DukeException If user input is an invalid string command.
+     */
+    public static Command parseInput(String userInput) throws DukeException {
+        String[] split = userInput.split(" ");
```
Do you think it's better if you add an additional line `String command = split[0]` here? That way, you can use `command.equals(xxx)` rather than `split[0].equals(xxx)`, which I think is more readable 🤣 .

njxue commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,107 @@
+     * @throws DukeException If user input is an invalid string command.
+     */
+    public static Command parseInput(String userInput) throws DukeException {
+        String[] split = userInput.split(" ");
```
Do you think it's better if you add an additional line `String command = split[0]` here? That way, you can use `command.equals(xxx)` rather than `split[0].equals(xxx)`, which I think is more readable 🤣 .

dexter-sim replied:
It is a good idea. I will implement your suggestion. Thank you for your review!

njxue reviewed (commented):
LGTM!

dexter-sim committed:
Fix text overflow bug

24Donovan24 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,107 @@
+     */
+    public static Command parseInput(String userInput) throws DukeException {
+        String[] split = userInput.split(" ");
+        if (userInput.equals("welcome") && split.length == 1) {
```
Do you think it will be neater to implement the conditionals as a switch case rather than multiple ifs statements?

24Donovan24 commented on src/main/java/duke/Ui.java:
```diff
@@ -20,109 +18,84 @@ public class Ui {
-    }
-
     /**
      * Prints Duke logo and greeting message.
```
JavaDocs can include the return type for some of the methods here

24Donovan24 reviewed (commented):
LGTM! Just some minor nits to fix!

turretDive commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,76 @@
+     * @param keyword A string to be checked.
+     * @return If description contains specified keyword.
+     */
+    public boolean match(String keyword) {
```
Maybe it would be good to named Boolean methods to sound like booleans? Such as canMatch() or isMatch().

turretDive commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,132 @@
+     *
+     * @return Size of the list.
+     */
+    public int size() {
```
Maybe it would be good to use verbs for Actions. Such as getSize().

turretDive commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    @Override
+    public String stringify() {
```
Adding a comment here on the difference between stringify() and toString() such as the difference in uses will explain why there is a need for another similar function 👍

turretDive reviewed (commented):
LGTM! Overall just come nits found in naming convention and documentation etc. I really liked how clean the overall code looked!

dexter-sim committed:
Update javadoc for refactored methods in ui

Ui methods: documentation are missing @return statements.

There are several methods that were refactored for the GUI and they return a string now.

Adding Javadoc return statements to complete documentation.

dexter-sim committed:
Improve code quality and fix coding standard violations

Some methods do not obey coding standard.

These methods were identified through peer review from the pull request.

Removing magic numbers and strings help improve maintainability.
Renaming methods to better represent the purpose of them.

Refactoring all those methods improves the code quality.

dexter-sim committed:
Improve code quality

Code abstraction is limited, especially in Parser class.

Methods gets too long and becomes difficult to maintain and extend.

Implement some of the code quality techniques to make code more organized.

Let's apply code quality guidelines and principles to all classes wherever necessary.

dexter-sim committed:
Refactor status icon of tasks

dexter-sim committed:
Add assert at various points in the code

dexter-sim committed:
Merge pull request #1 from dexter-sim/branch-A-Assertions

Add assert at various points in the code

dexter-sim committed:
Merge branch 'master' into branch-A-CodeQuality

dexter-sim committed:
Merge pull request #2 from dexter-sim/branch-A-CodeQuality

Branch a code quality

dexter-sim committed:
Add ability to recognize duplicate tasks

dexter-sim committed:
Merge pull request #3 from dexter-sim/branch-C-DetectDuplicates

Add ability to recognize duplicate tasks

dexter-sim committed:
Add a user guide to the project

dexter-sim committed:
Update feature list in user guide

dexter-sim committed:
Fix test cases

dexter-sim committed:
Reorder import statements
